tincr/io/device/xdlrc.tcl

```diff
@@ -154,7 +154,8 @@ proc ::tincr::write_xdlrc { args } {
             
             # For ultrascale devices add power/ground source sites that aren't explicitly represented in Vivado
             if {$is_series7 == 0} {
-                lappend site_types "VCC" "GND"
+                dict lappend site_types "VCC" "NULL"
+                dict lappend site_types "GND" "NULL"
```

Comment:
Why are we populating `site_types` with the `::tincr::sites unique` command? We should be using the `::tincr::sites get_types` command instead. Per the documentation:

> [The `::tincr::sites unique` command creates] a dictionary that maps a site type to a physical site location on the current device.

Yet we never use the pointer to the physical site location in `write_xdlrc`. The reason you have to add the `"NULL"` is because you're required to add a value for every key in a dictionary.

The `::tincr::sites get_types` command returns a sorted list instead of a dictionary, is faster, and provides just the information you need:

> [The `::tincr::sites get_types` command returns] a list of all site types in the current device.